homeassistant/components/rachio/binary_sensor.py

```diff
@@ -43,6 +46,68 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True, kw_only=True)
+class RachioControllerBinarySensorDescription(BinarySensorEntityDescription):
+    """Describe a Rachio controller binary sensor."""
+
+    update_received: Callable[[str], bool | None]
+    is_on: Callable[[RachioIro], bool]
+    signal_string: str
+
+
+CONTROLLER_BINARY_SENSOR_TYPES: tuple[RachioControllerBinarySensorDescription, ...] = (
+    RachioControllerBinarySensorDescription(
+        key=KEY_ONLINE,
+        device_class=BinarySensorDeviceClass.CONNECTIVITY,
+        signal_string=SIGNAL_RACHIO_CONTROLLER_UPDATE,
+        is_on=lambda controller: controller.init_data[KEY_STATUS] == STATUS_ONLINE,
+        update_received=lambda state: True
+        if state in (SUBTYPE_ONLINE, SUBTYPE_COLD_REBOOT)
+        else False
+        if state == SUBTYPE_OFFLINE
+        else None,
```

Comment:
Please wrap multiline lambdas in parenthesis

Reply:
For that matter this should work here then shouldn't it?

```
 RachioControllerBinarySensorDescription(
        key=KEY_ONLINE,
        device_class=BinarySensorDeviceClass.CONNECTIVITY,
        signal_string=SIGNAL_RACHIO_CONTROLLER_UPDATE,
        is_on=lambda controller: controller.init_data[KEY_STATUS] == STATUS_ONLINE,
        update_received={
            SUBTYPE_ONLINE: True,
            SUBTYPE_COLD_REBOOT: True,
            SUBTYPE_OFFLINE: False,
        }.get,
```